Objects/genericaliasobject.c

```diff
@@ -626,6 +640,65 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    gaobject *alias = (gaobject *) gi->obj;
+
+    if (gi->exhausted) {
+        gi->obj = NULL;
+        return NULL;
```

Comment:
I feel like you could potentially be missing a decref here for `gi->obj` since I see it being increfed in `gi_iter`. If so,
```suggestion
        Py_SETREF(gi->obj, NULL);
        return NULL;
```
If I'm just shooting my mouth off, then just please ignore what I said :).

One way to test for refleaks is to run the tests with the refleak hunter (requires a debug build):
```
python_d.exe -m test test_typing -R 3:3
```